tests/test_association.py

```diff
@@ -0,0 +1,357 @@
+
+# Define the root of the tests relative to this file
+ROOT = os.path.abspath(os.path.dirname(__file__))
+
+# Define a generic dataId
+dataId_dict = {'visit': 1111,
+               'filter': 'r'}
+
+
+def create_test_dia_objects(n_objects=1,
+                            n_sources=1,
+                            start_id=0,
+                            object_centers_degrees=None,
+                            scatter_arcsec=1.0):
+    """ Create DIAObjects with a specified number of DIASources attached.
+
+    Parameters
+    ----------
+    n_objects : int
+        Number of DIAObjects to generate.
+    n_src : int
+        Number of DIASources to generate for each DIAObject.
+    start_id : int
+        Starting index to increment the created DIAObjects from.
+    object_centers_degrees : (N, 2) list of floats
+        Centers of each DIAObject to create.
+    scatter_arcsec : float
+        Scatter to add to the position of each DIASource.
+
+    Returns
+    -------
+    A list of DIAObjects
+    """
+
+    if object_centers_degrees is None:
+        object_centers_degrees = [[idx, idx] for idx in range(n_objects)]
+
+    output_dia_objects = []
+    for obj_idx in range(n_objects):
+        src_cat = create_test_sources(
+            n_sources,
+            start_id + obj_idx * n_sources,
+            [object_centers_degrees[obj_idx] for src_idx in range(n_sources)],
+            scatter_arcsec)
+        output_dia_objects.append(DIAObject(src_cat))
+    return output_dia_objects
+
+
+def create_test_sources(n_sources=5,
+                        start_id=0,
+                        source_locs_deg=None,
+                        scatter_arcsec=1.0):
+    """ Create dummy DIASources for use in our tests.
+
+    Parameters
+    ----------
+    n_sources : int
+        Number of fake sources to create for testing.
+    start_id : int
+        Unique id of the first object to create. The remaining sources are
+        incremented by one from the first id.
+    source_locs_deg : (N, 2) list of floats
+        Positions of the DIASources to create.
+    scatter_arcsec : float
+        Scatter to add to the position of each DIASource.
+
+    Returns
+    -------
+    A lsst.afw.SourceCatalog
+    """
+    sources = afwTable.SourceCatalog(make_minimal_dia_source_schema())
+
+    if source_locs_deg is None:
+        source_locs_deg = [[idx, idx] for idx in range(n_sources)]
+
+    for src_idx in range(n_sources):
+        src = sources.addNew()
+        src['id'] = src_idx + start_id
+        coord = Coord(source_locs_deg[src_idx][0] * afwGeom.degrees,
+                      source_locs_deg[src_idx][1] * afwGeom.degrees)
+        if scatter_arcsec > 0.0:
+            coord.offset(
+                np.random.rand() * 360 * afwGeom.degrees,
+                np.random.rand() * scatter_arcsec * afwGeom.arcseconds)
+        src.setCoord(coord)
+
+    return sources
+
+
+class MeasureAssociationTestSuite(lsst.utils.tests.TestCase):
+
+    def setUp(self):
```

Comment:
Setting up a butler for tests can sometimes be very slow. If that's the case here, it might be better to use `setUpClass` instead for some of the setup.

Reply:
It doesn't seem to be currently. The tests take overall 2 seconds.